src/jshint.js

```diff
@@ -3754,18 +3754,25 @@ var JSHINT = (function() {
     var b;
 
     function doCatch() {
-      var e;
+      var params = [];
 
       advance("catch");
       advance("(");
 
-      e = state.tokens.next;
-      if (e.type !== "(identifier)") {
-        warning("E030", state.tokens.next, e.value);
-        e = null;
+      if (checkPunctuators(state.tokens.next, ["[", "{"])) {
+        var tokens = destructuringExpression();
+        var t;
+        for (t in tokens) {
```

Comment:
I don't like the unguarded for-in, but that's the way its already done elsewhere. Maybe I'm just old fashioned. Who extends the array prototype now-a-days?